### DISCUSSION TITLE
using `ysw<char>` instead of `ysiw<char>`

### DISCUSSION THREAD

**Discussion by azinsharaf** [kylechui/nvim-surround#257] [General]
for adding i use `ysiw<char>`. Why do we need to type `i`? Can't i just use `ysw<char>` ? How can i add this keymap to my config?

**Comment by kylechui** [comment]
You don't *need* to use `iw`, it just happens to be the motion that selects the entire word the cursor is on, regardless of where the cursor is. If you just use the `w` motion, that will select *from the cursor* to the end of the word. You can try remapping the motion using `onoremap` (not really recommended), or try remapping `ysw` to `<Plug>(nvim-surround-normal)iw` (not 100% sure if that syntax is right, check `:h nvim-surround` to make sure).

**Reply by azinsharaf** [reply]
thanks for the help. It worked! 

```
return {
    "kylechui/nvim-surround",
    version = "*", -- Use for stability; omit to use `main` branch for the latest features
    event = "VeryLazy",
    config = function()
        require("nvim-surround").setup({
            -- Configuration here, or leave empty to use defaults
			keymaps = {
			ysw = "<Plug>(nvim-surround-normal)iw",
			},
        })
    end
}
```

**Reply by kylechui** [reply]
That's actually not the intended use-case for the `keymaps` field; I would recommend you move the keymap outside of `nvim-surround`'s `setup` function, i.e. into a separate `vim.keymap.set` call.
